### ISSUE TITLE
Recursion error - RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by bwnance** [microsoft/pyright#4798]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
Pyright crashes when evaluated on the attached class.

<details>
  <summary>Logs Here</summary>

  ```
  Assuming Python platform Darwin
  Searching for source files
  Found 1 source file
  An internal error occurred while type checking file "/Users/brandon.nance/tracer/backend/utils/chor_typetest.py": RangeError: Maximum call stack size exceeded
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9452:54)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3718:39)
      at doForEachSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:684:9)
      at expandSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3711:13)
      at forEach (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3739:17)
      at Array.forEach (<anonymous>)
      at mapSubtypesExpandTypeVars (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3738:27)
      at validateCallArguments (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9011:26)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9442:56)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3718:39)
      at doForEachSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:684:9)
      at expandSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3711:13)
      at forEach (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3739:17)
      at Array.forEach (<anonymous>)
      at mapSubtypesExpandTypeVars (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3738:27)
      at validateCallArguments (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9011:26)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9442:56)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3718:39)
      at doForEachSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:684:9)
      at expandSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3711:13)
      at forEach (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3739:17)
      at Array.forEach (<anonymous>)
      at mapSubtypesExpandTypeVars (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3738:27)
      at validateCallArguments (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9011:26)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9442:56)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3718:39)
      at doForEachSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:684:9)
      at expandSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3711:13)
      at forEach (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3739:17)
      at Array.forEach (<anonymous>)
      at mapSubtypesExpandTypeVars (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3738:27)
      at validateCallArguments (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9011:26)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9442:56)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3718:39)
      at doForEachSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:684:9)
      at expandSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3711:13)
      at forEach (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3739:17)
      at Array.forEach (<anonymous>)
      at mapSubtypesExpandTypeVars (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3738:27)
      at validateCallArguments (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9011:26)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9442:56)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3718:39)
      at doForEachSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:684:9)
      at expandSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3711:13)
      at forEach (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3739:17)
      at Array.forEach (<anonymous>)
      at mapSubtypesExpandTypeVars (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3738:27)
      at validateCallArguments (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9011:26)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9442:56)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3718:39)
      at doForEachSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:684:9)
      at expandSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3711:13)
      at forEach (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3739:17)
      at Array.forEach (<anonymous>)
      at mapSubtypesExpandTypeVars (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3738:27)
      at validateCallArguments (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9011:26)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9442:56)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3718:39)
      at doForEachSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:684:9)
      at expandSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3711:13)
      at forEach (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3739:17)
      at Array.forEach (<anonymous>)
      at mapSubtypesExpandTypeVars (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3738:27)
      at Ht (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9011:26)
  Error performing analysis: RangeError: Maximum call stack size exceeded
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9452:54)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3718:39)
      at doForEachSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:684:9)
      at expandSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3711:13)
      at forEach (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3739:17)
      at Array.forEach (<anonymous>)
      at mapSubtypesExpandTypeVars (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3738:27)
      at validateCallArguments (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9011:26)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9442:56)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3718:39)
      at doForEachSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:684:9)
      at expandSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3711:13)
      at forEach (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3739:17)
      at Array.forEach (<anonymous>)
      at mapSubtypesExpandTypeVars (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3738:27)
      at validateCallArguments (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9011:26)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9442:56)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3718:39)
      at doForEachSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:684:9)
      at expandSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3711:13)
      at forEach (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3739:17)
      at Array.forEach (<anonymous>)
      at mapSubtypesExpandTypeVars (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3738:27)
      at validateCallArguments (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9011:26)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9442:56)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3718:39)
      at doForEachSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:684:9)
      at expandSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3711:13)
      at forEach (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3739:17)
      at Array.forEach (<anonymous>)
      at mapSubtypesExpandTypeVars (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3738:27)
      at validateCallArguments (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9011:26)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9442:56)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3718:39)
      at doForEachSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:684:9)
      at expandSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3711:13)
      at forEach (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3739:17)
      at Array.forEach (<anonymous>)
      at mapSubtypesExpandTypeVars (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3738:27)
      at validateCallArguments (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9011:26)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9442:56)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3718:39)
      at doForEachSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:684:9)
      at expandSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3711:13)
      at forEach (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3739:17)
      at Array.forEach (<anonymous>)
      at mapSubtypesExpandTypeVars (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3738:27)
      at validateCallArguments (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9011:26)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9442:56)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3718:39)
      at doForEachSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:684:9)
      at expandSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3711:13)
      at forEach (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3739:17)
      at Array.forEach (<anonymous>)
      at mapSubtypesExpandTypeVars (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3738:27)
      at validateCallArguments (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9011:26)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9442:56)
      at callback (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3718:39)
      at doForEachSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:684:9)
      at expandSubtype (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3711:13)
      at forEach (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3739:17)
      at Array.forEach (<anonymous>)
      at mapSubtypesExpandTypeVars (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3738:27)
      at Ht (/Users/brandon.nance/.cache/pyright-python/1.1.298/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9011:26)
  ```

</details>

**To Reproduce**
put attached code into class, run pyright on it.

**Expected behavior**
Though this is a strange thing to do, i would not expect pyright to crash when analyzing

**Screenshots or Code**
```python
from typing import Any


from pandas.core.groupby.generic import DataFrameGroupBy as DataFrameGroupByType
from pandas.core.frame import DataFrame as DataFrameType


class ChorObject:
    SUPPORTED_INDEXERS = [
        "loc",
        "iloc",
        "at",
        "iat",
    ]

    def __init__(self, pd_object):
        self._pd = pd_object

    def __getattr__(self, name):
        return self.generic(name)

    def generic(self, name):

        val = getattr(self._pd, name)
        if name in self.SUPPORTED_INDEXERS or not callable(val):
            return ChorUtils.coerce_chor(val)

        def wraps(*args, **kwargs):
            args = [ChorUtils.coerce_pandas(i) for i in args]
            val = getattr(self._pd, name)(*args, **kwargs)
            return ChorUtils.coerce_chor(val)

        return wraps


class ChorFrame(ChorObject):

    _TYPE = DataFrameType

    def __init__(  # NOQA MC0001
        self,
        dataframe=None,
    ):

        if isinstance(dataframe, self.__class__):
            dataframe = dataframe._pd

        super().__init__(dataframe)


class ChorGroup(ChorObject):
    _TYPE = DataFrameGroupByType

    def __init__(self, groupby):
        self.groupby = groupby


class ChorUtils:
    @classmethod
    def coerce_chor(cls, val):
        # note: using an Any return typehint fixes pyright recursion bug
        #   that causes it to crash on any file that uses chor.
        #   though this func _should_ be -> Union[ChorFrame, ChorGroup]

        if isinstance(val, ChorFrame._TYPE):
            return ChorFrame(dataframe=val)
        elif isinstance(val, ChorGroup._TYPE):
            return ChorGroup(val)
        return val

    @classmethod
    def coerce_pandas(cls, val):
        pass

    @staticmethod
    def memory_mb(pandas_obj) -> int:
        if isinstance(pandas_obj, (ChorFrame._TYPE, ChorFrame)):
            usage_b = pandas_obj.memory_usage(deep=True).sum()
        else:  # we assume if not a df it's a series
            usage_b = pandas_obj.memory_usage(deep=True)
        usage_mb = usage_b / 1024**2  # convert bytes to megabytes
        return round(usage_mb, 2)
```

**VS Code extension or command-line**
happens regardless of whether it's an extension (pyright or pylance) or not - for testing, i've been using pyright in the cli, version `1.1.298`

**Additional context**
The sample code is a very cut down version of a library that wraps pandas (yes, i know something like what this class does is a weird thing to do).
the issue appears to originate from the `coerce_chor` function, and the use of overriding `__getattr__` via `ChorObject`. if i add a return typehint of  `Any` to the `coerce_chor` function, pyright no longer crashes and behaves just fine.

the concept behind the class and the __getattr__ stuff, if it helps, is to allow the override of certain pandas functions in our class, but pass through others to the standard pandas interface, when they aren't defined

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by bwnance** [comment]
thanks so much! appreciate the prompt response :)

**Comment by erictraut** [comment]
This is included in pyright 1.1.300, which I just published. It will also be included in a future release of pylance.
